assets/src/components/searchForm.tsx

```diff
@@ -146,25 +160,93 @@ const Filters = ({
   )
 }
 
+export const MapSearchCombobox = ({
+  autocompleteController,
+  autocompleteId,
+  dispatch,
+  formSearchInput,
+  onSubmit,
+  query,
+  setAutocompleteEnabled,
+}: {
+  autocompleteController: MutableRefObject<GroupedAutocompleteControls | null>
+  autocompleteId: string
+  dispatch: Dispatch
+  formSearchInput: React.MutableRefObject<HTMLInputElement | null>
+  onSubmit: (e: SyntheticEvent) => void
+  query: SearchQuery
+  setAutocompleteEnabled: React.Dispatch<React.SetStateAction<boolean>>
+}) => {
+  const onSelectVehicleOption = (vehicle: Vehicle | Ghost) => {
+    dispatch(
+      setSelectedEntity({
+        type: SelectedEntityType.Vehicle,
+        vehicleId: vehicle.id,
+      })
+    )
+  }
+
+  const onSelectedLocationId = (id: string) => {
+    dispatch(
+      setSelectedEntity({
+        type: SelectedEntityType.LocationByPlaceId,
+        placeId: id,
+      })
+    )
+  }
+
+  const onSelectedLocationText = (text: string) => {
+    dispatch(setSearchText(text))
+    dispatch(submitSearch())
+  }
+
+  return (
+    <SocketContext.Provider value={useSocket()}>
+      <GroupedAutocompleteFromSearchTextResults
+        id={autocompleteId}
+        controlName="Search Suggestions"
+        maxElementsPerGroup={5}
+        searchFilters={
+          query.property === "all"
+            ? allFiltersOn
+            : { ...allFiltersOff, [query.property]: true }
+        }
+        searchText={query.text}
+        fallbackOption={autocompleteOption(query.text, onSubmit)}
+        onSelectVehicleOption={onSelectVehicleOption}
+        onSelectedLocationId={onSelectedLocationId}
+        onSelectedLocationText={(text) => {
+          setAutocompleteEnabled(false)
+          onSelectedLocationText(text)
+        }}
+        controllerRef={autocompleteController}
+        onCursor={{
+          onCursorExitEdge: () => formSearchInput.current?.focus(),
+        }}
+      />
+    </SocketContext.Provider>
+  )
+}
+
 /**
- * Search form which exposes all configurable state and callbacks via {@link SearchFormProps props}.
+ * Search form which exposes all configurable state and callbacks via {@link ComboboxProps props}.
  */
-export const SearchForm = ({
+export const Combobox = ({
   inputText,
   onInputTextChange,
-  property,
-  onPropertyChange,
   onClear: onClearProp,
   onSubmit: onSubmitProp,
-  onSelectVehicleOption,
-  onSelectedLocationId,
-  onSelectedLocationText,
+  comboboxType,
   showAutocomplete: showAutocompleteProp = true,
-}: SearchFormProps) => {
+  dispatch,
+  query,
+  options,
+}: SearchComboboxProps | SelectComboboxProps) => {
   const formSearchInput = useRef<HTMLInputElement | null>(null)
   const [autocompleteEnabled, setAutocompleteEnabled] = useState(true)
 
   const onSubmit = (e: SyntheticEvent) => {
+    e.preventDefault()
```

Comment:
question: do any tests fail if this is removed, or is any strange behavior introduced?

I'm curious because I wonder if this was something added when initially working on the TODO from this resolved comment https://github.com/mbta/skate/pull/2659/files/671ac7f7775f73b7f349d4679412442b7d279b9a#r1638262192

Reply:
Good memory, totally right